bot/utils/helpers.py

```diff
@@ -6,3 +6,30 @@ def suppress_links(message: str) -> str:
     for link in set(re.findall(r"https?://[^\s]+", message, re.IGNORECASE)):
         message = message.replace(link, f"<{link}>")
     return message
+
+def neutralise_string(txt: str | None) -> str | None:
+    """Attempts to neutralise all punctuation and cases and returns a string of lowercase words."""
+    # return early if no text provided.
+    if not txt:
+        return None
+
+    # take out punctuation
+    txt = re.sub(r"([^\w\s]|_)", " ", txt)
```

Comment:
This makes it clearer and more readable IMO. It's entirely up to you to accept this suggestion.
```suggestion
    # Split by underscore and non-alphanumeric characters
    txt = re.sub(r"[\W_]", " ", txt)
```